### ISSUE TITLE
queryRenderedFeatures returns wrong results

### ISSUE THREAD

**Issue by moodstubos** [maplibre/maplibre-gl-native#184]
queryRenderedFeatures does not work anymore. wrong - incomprehensible - results. 

@ntadej
issue startet with this commit: https://github.com/maplibre/maplibre-gl-native/commit/fe041b80caca953d5ffbea19b5d2a36484d0286a

before this commit (e.g. https://github.com/maplibre/maplibre-gl-native/commit/c0d76c69cc6f3abbc2be44c79f387bab114a0875) everything is as expected.

**Comment by ntadej** [comment]
Can you try to make a minimal reproducer? I see there's a unit test available, but not sure how to run it for Android: https://github.com/maplibre/maplibre-gl-native/blob/master/test/api/query.test.cpp

**Comment by moodstubos** [comment]
After some (more) tests I find out that this change causes the problems:

src/mbgl/util/tile_coordinate.hpp 
https://github.com/maplibre/maplibre-gl-native/commit/fe041b80caca953d5ffbea19b5d2a36484d0286a#diff-0ee83ffd64534bebe9fbe1eff7412f516881e00460b30a21ba4edefae1830441

After reverse this change and recompile all seems to be like it should be.

I am absolutly not familiar with C++ and cannot see the issue. Perhaps anybody can comprehend (and explain) why this change makes problems.

**Comment by moodstubos** [comment]
oh sorry, 
could it be that this commit is relating and fixed it?
135dca679b648f885b9c7273c4b820faf597c826 

I only tested latest Android release https://github.com/maplibre/maplibre-gl-native/releases/tag/android-v9.5.1 and not latest MASTER

If this is the case and this issue does not exist in master i think there should be a hotfix release android-v9.5.2 in order to fix this  critical issue in public release version.

**Comment by mkieselmann** [comment]
I have the same issue with the latest iOS release v5.12.1. 

Going back to v5.12.0 resolves the issue and `MapView.visibleFeatures(at: CGPoint)` works just fine.

**Comment by moodstubos** [comment]
@mkieselmann 
yes. I can confirm commit [fe041b8](https://github.com/maplibre/maplibre-gl-native/commit/fe041b80caca953d5ffbea19b5d2a36484d0286a) has also landed in ios-5.12.1 release. So you have to wait for a new release (or compile master yourself). 
unfortunately i don't know if this project is still alive and if further releases will come.

**Comment by petr-pokorny-1** [comment]
@mkieselmann @moodstubos @ntadej I have created [new android release 9.5.2](https://github.com/maplibre/maplibre-gl-native/releases/tag/android-v9.5.2). Any chance to re-test this issue with the new release so that we can close it? Thanks!
